### PR TITLE
[develop] export LayoutGraphData to editor

### PR DESCRIPTION
**Pull request by star-e**
export layout graph data to editor.

### Changelog

* simplified graph implementation
* reordered files in CMakeLists.txt

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

**Comment by star-e** [comment]
@cocos-robot run test cases

**Comment by star-e** [comment]
@cocos-robot run test cases

**Comment by star-e** [comment]
@cocos-robot run test cases

File: editor/exports/custom-pipeline.ts [hunk 1]
```diff
@@ -0,0 +1,2 @@
+import * as shader from '../../cocos/rendering/custom/editor';
+export { shader };
```
Comment (star-e) [inline editor/exports/custom-pipeline.ts:2]: Add types to cc.shader

**Review by dumganhar — APPROVED** [review]

**Pull request merged**